marklogic-data-hub/src/main/resources/ml-modules/root/marklogic-data-hub-spark-connector/partition-lib.xqy

```diff
@@ -0,0 +1,172 @@
+xquery version "1.0-ml";
+
+module namespace partition = "org:example:partition";
+
+import module namespace op = "http://marklogic.com/optic" at "/MarkLogic/optic.xqy";
+import module namespace ofn = "http://marklogic.com/optic/expression/fn" at "/MarkLogic/optic/optic-fn.xqy";
+import module namespace osql = "http://marklogic.com/optic/expression/sql" at "/MarkLogic/optic/optic-sql.xqy";
+
+(:
+Returns an "initial" set of partitions that just define a min and max for each partition. This is implemented in
+XQuery to avoid any problems with handling unsignedLong's.
+:)
+declare function build-initial-partitions($partition-count as xs:integer) as json:array
```

Comment:
@ehennum If you have a moment, was hoping you could check my XQuery here. There are unit tests below to ensure that the property min/max values are being calculated, just want to make sure nothing looks off.